src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int RANDOM_FIGURE_NUMBER = 5;
+    private static final int RANDOM_SIDE_LENGTH = 10;
+    private ColorSupplier supplier = new ColorSupplier();
+
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int randomFigure = random.nextInt(RANDOM_FIGURE_NUMBER);
+        String colore = supplier.getRandomColor();
+        int randomSide = getRandomSide();
+        Figure circle = new Circle(randomSide, colore);
+        Figure square = new Square(randomSide, colore);
+        Rectangle rectangle = new Rectangle(randomSide, randomSide, colore);
+        RightTriangle rightTriangle = new RightTriangle(randomSide, randomSide, colore);
+        IsoscelesTrapezoid isoscelesTrapezoid = new IsoscelesTrapezoid(randomSide, randomSide,
+                randomSide, colore);
+
+        switch (randomFigure) {
+            case 0:

```

Comment:
You can use the latest version of switch case:

return switch (randomFigure) {
case 0 -> new Circle(getRandomSide(), color);
case 1 ->  new Square(getRandomSide(), color);
....
default ->  new IsoscelesTrapezoid(getRandomSide(), getRandomSide(),
                getRandomSide(), color);
};